crates/oxc_linter/src/rules/eslint/no_useless_escape.rs

```diff
@@ -32,6 +32,10 @@ declare_oxc_lint!(
 
 impl Rule for NoUselessEscape {
     fn run<'a>(&self, node: &AstNode<'a>, ctx: &LintContext<'a>) {
+        if matches!(ctx.nodes().parent_kind(node.id()), Some(AstKind::JSXAttributeItem(_))) {
```

Comment:
I think we should do this check just before raising an error, for better performance. Remember, `run` is called on every single AST node so we want to do as little work as possible up front.

Reply:
good point